modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/product/navigation/control/menu/FragmentEntriesGeneratorProductNavigationControlMenuEntry.java

```diff
@@ -86,12 +87,11 @@ public boolean isShow(HttpServletRequest httpServletRequest)
 	}
 
 	@Override
-	@Reference(
-		target = "(osgi.web.symbolicname=com.liferay.fragment.web)",
-		unbind = "-"
-	)
-	public void setServletContext(ServletContext servletContext) {
-		super.setServletContext(servletContext);
+	protected RequestDispatcher getRequestDispatcher(String jspPath) {
```

Comment:
@W2jx Please make the method to be getServletContext, then you can move the duplicate logic _servletContext.getRequestDispatcher(jspPath); into its base class

Reply:
I will fix it.